ci-operator/config/openshift/installer/openshift-installer-master__terraform.yaml

```diff
@@ -0,0 +1,110 @@
+    COPY . .
+    RUN CGO_ENABLED=0 GOOS=darwin GOARCH=arm64 DEFAULT_ARCH="$(go env GOHOSTARCH)" make -C terraform all
+    RUN find /go/src/github.com/openshift/installer/terraform/bin -executable ! -name "terraform" -type f -exec rm -f {} \;
+
+    FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.14 AS linuxbuilder
+    ARG TAGS=""
+    WORKDIR /go/src/github.com/openshift/installer
+    COPY . .
+    RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 DEFAULT_ARCH="$(go env GOHOSTARCH)" make -C terraform all
+    RUN find /go/src/github.com/openshift/installer/terraform/bin -executable ! -name "terraform" -type f -exec rm -f {} \;
+
+    FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.14 AS linuxarmbuilder
+    ARG TAGS=""
+    WORKDIR /go/src/github.com/openshift/installer
+    COPY . .
+    RUN CGO_ENABLED=0 GOOS=linux GOARCH=arm64 DEFAULT_ARCH="$(go env GOHOSTARCH)" make -C terraform all
+    RUN find /go/src/github.com/openshift/installer/terraform/bin -executable ! -name "terraform" -type f -exec rm -f {} \;
+
+    FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.14 AS builder
+    WORKDIR /go/src/github.com/openshift/installer
+    COPY --from=macbuilder /go/src/github.com/openshift/installer/terraform/bin/ terraform/bin/
+    COPY --from=macarmbuilder /go/src/github.com/openshift/installer/terraform/bin/ terraform/bin/
+    COPY --from=linuxbuilder /go/src/github.com/openshift/installer/terraform/bin/ terraform/bin/
+    COPY --from=linuxarmbuilder /go/src/github.com/openshift/installer/terraform/bin/ terraform/bin/
+  from: base
+  inputs:
+    ocp_builder_rhel-8-golang-1.18-openshift-4.12:
+      as:
+      - registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.12
+    ocp_builder_rhel-8-golang-1.19-openshift-4.14:
+      as:
+      - registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.19-openshift-4.14
+    ocp_builder_rhel-8-golang-1.20-openshift-4.14:
+      as:
+      - registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.14
+  to: terraform-artifacts
+promotion:
+  name: "4.14"
+  namespace: ocp
+releases:
+  initial:
+    integration:
+      name: "4.14"
+      namespace: ocp
+  latest:
+    integration:
+      include_built_images: true
+      name: "4.14"
+      namespace: ocp
+resources:
+  '*':
+    requests:
+      cpu: 100m
+      memory: 200Mi
+  terraform-artifacts:
+    requests:
+      cpu: "3"
+      memory: 5Gi
+tests:
+- as: tf-check
+  commands: |
+    function fail() {
+      echo "$1"
+      exit 1
+    }
+    nplatforms=4 # (linux|darwin)_(amd|arm)64
+    [ $(ls terraform/bin/ | wc -w) == $nplatforms ] || fail "Artifacts were not generated for all platforms"
+    nbins=17 # 16 providers + terraform bin itself
+    for dir in terraform/bin/*; do
+      [ $(ls $dir | wc -w) == $nbins ] || fail "Missing binaries on $dir"
+    done
+  container:
+    from: terraform-artifacts
+  run_if_changed: terraform
```

Comment:
As per the [Prow docs](https://docs.prow.k8s.io/docs/jobs/#triggering-jobs-based-on-changes), `run_if_changed` is matched against the path of each changed file (and it's sufficient that _any_ will match). The current form should work, but just to be a slightly bit safer side we could use `  run_if_changed: ^terraform/.*`